use_cases/eluc/app/app.py

```diff
@@ -1,684 +1,62 @@
-    :return: Slider values scaled down to fit percentage of land used in context.
-    """
-    context = df.loc[year, lat, lon]
-    presc = pd.Series(sliders, index=constants.RECO_COLS)
+legend_component = LegendComponent()
 
-    old_sum = context[constants.RECO_COLS].sum()
-    new_sum = presc.sum()
+map_component = MapComponent(df)
+map_component.register_update_map_callback(app)
+map_component.register_click_map_callback(app)
+map_component.register_select_country_callback(app)
 
-    # TODO: There is certainly a more elegant way to handle this.
-    if locked:
-        unlocked = [col for col in constants.RECO_COLS if col not in locked]
-        locked_sum = presc[locked].sum()
-        old_sum -= locked_sum
-        new_sum -= locked_sum
-        # We do this to avoid divide by zero. In the case where new_sum == 0
-        # we have all locked columns and/or zero columns so no adjustment is needed
-        if new_sum != 0:
-            presc[unlocked] = presc[unlocked].div(new_sum).mul(old_sum)
+prescription_component = PrescriptionComponent(df)
+prescription_component.register_select_prescriptor_callback(app)
+prescription_component.register_toggle_modal_callback(app)
 
-    else:
-        presc = presc.div(new_sum).mul(old_sum)
+sliders_component = SlidersComponent(df)
+sliders_component.register_set_frozen_reset_sliders_callback(app)
+sliders_component.register_show_slider_value_callback(app)
+sliders_component.register_sum_to_one_callback(app)
 
-    # Set all negative values to 0
-    presc[presc < 0] = 0
-    return presc.tolist()
+lock_component = LockComponent()
 
-@app.callback(
-    Output("predict-eluc", "value"),
-    Input("predict-button", "n_clicks"),
-    State("year-input", "value"),
-    State("lat-dropdown", "value"),
-    State("lon-dropdown", "value"),
-    State({"type": "presc-slider", "index": ALL}, "value"),
-    State("pred-select", "value"),
-    prevent_initial_call=True
-)
-def predict(_, year, lat, lon, sliders, predictor_name):
-    """
-    Predicts ELUC from context and prescription stores.
-    :param year: Selected context year.
-    :param lat: Selected context lat.
-    :param lon: Selected context lon.
-    :param sliders: Prescribed slider values.
-    :param predictor_name: String name of predictor to use from dropdown.
-    :return: Predicted ELUC.
-    """
-    context = df.loc[year, lat, lon]
-    presc = pd.Series(sliders, index=constants.RECO_COLS)
-    context_actions_df = utils.context_presc_to_df(context, presc)
+chart_component = ChartComponent(df)
+chart_component.register_update_context_chart_callback(app)
+chart_component.register_update_presc_chart_callback(app)
 
-    predictor = predictors[predictor_name]
-    eluc_df = predictor.predict(context_actions_df)
-    eluc = eluc_df["ELUC"].iloc[0]
-    return f"{eluc:.4f}"
+prediction_component = PredictionComponent(df)
+prediction_component.register_predictor_callback(app)
+prediction_component.register_land_use_callback(app)
```

Comment:
The idea behind these components is we instantiate them then call the register functions to enable callbacks.